### DISCUSSION TITLE
Jump to next file in the file tree (skip dir)

### DISCUSSION THREAD

**Discussion by wojciech-kulik** [jesseduffield/lazygit#4069] [Q&A]
I like the file tree, but it's quite annoying that when you press the down arrow you also are able to select a directory that shows a diff for the whole subtree. So when I'm reviewing the code and hit `down`, `down`... I have to look at the tree to see when to skip the dir. 

Of course, I could switch to flat list, but as I said I would prefer to have tree view wit the ability to jump between files and skipping dirs. Is it possible?

**Comment by stefanhaller** [comment]
When you say your goal is to only jump between files without having to look at the Files view, then I don't understand why switching to the flat view isn't the best solution for that. Hitting backspace to switch between flat view and tree view is very easy, and it even retains the selection. Seems perfect to me.

Modifying up/down in tree view to skip directories is not an option, of course. It is important to be able to select directories in order to collapse/expand them; also, I often want to see the combined diff of all files in a directory, because I'm using the git option `diff.colorMoved` which shows me how code moved from one file to another, and whether it was modified along the way. This is not possible when single files are selected (unless the code moved within a single file), so that's why I tend to select directories more often then single files when reviewing my code.

One option might be to add additional keybindings for next/previous file that skip directories, but I can't think of good keys to use for that.

**Reply by wojciech-kulik** [reply]
>  I don't understand why switching to the flat view isn't the best solution for that

Because then I don't see where I am in the hierarchy, so it would require switching back and forth between those two views. My use case is that I do self code review before pushing changes and I want to switch efficiently between files while seeing folder hierarchy for a better understanding of where I am. Paths displayed in a flat view are not very readable.

> One option might be to add additional keybindings for next/previous file that skip directories

That's what I was thinking about, I don't want to disable directory selection completely.

**Reply by stefanhaller** [reply]
Ok, I see. Any suggestions which keys to use? One option that comes to mind is ctrl-j/ctrl-k (as in "a variation of j/k"). Downsides are that it is not very similar to what those keys do in the commits panel, and they are probably too cumbersome to type (need two hands). Any other ideas?

**Reply by wojciech-kulik** [reply]
Maybe by default `ctrl+j` & `ctrl+k` but it would be nice to have it configurable. I could then for example remap it to arrow up/down. 

Or maybe it could be enabled like a toggle? After hitting some shortcut the default jumping would be changed (but it would be probably more difficult to implement, because it would also require some indication on UI showing in which mode we are).

**Reply by mark2185** [reply]
I'd prefer for it to be a key, not a toggle.

Imagine wanting to go to the next folder (and then browse the files in that folder), you'd need 3 key presses just to use it once.

**Reply by wojciech-kulik** [reply]
Yeap, a key is a better solution, preferably a configurable one.
